components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -262,22 +302,31 @@ void ConversationHandler::OnAssociatedContentDestroyed(
   // using our current cached content.
   associated_content_delegate_ = nullptr;
   if (!chat_history_.empty() && should_send_page_contents_ &&
-      associated_content_info_ && associated_content_info_->url.has_value()) {
+      metadata_->associated_content &&
+      metadata_->associated_content->url.has_value()) {
     // Get the latest version of article text and
     // associated_content_info_ if this chat has history and was connected to
-    // the associated conversation, then construct a "content archive"
-    // implementation of AssociatedContentDelegate with a duplicate of the
-    // article text.
-    auto archive_content = std::make_unique<AssociatedArchiveContent>(
-        associated_content_info_->url.value_or(GURL()), last_text_content,
-        base::UTF8ToUTF16(associated_content_info_->title.value_or("")),
-        is_video);
-    associated_content_delegate_ = archive_content->GetWeakPtr();
-    archive_content_ = std::move(archive_content);
+    // the associated conversation, then store the content so the conversation
+    // can continue.
+    SetArchiveContent(std::move(last_text_content), is_video);
   }
   OnAssociatedContentInfoChanged();
 }
 
+void ConversationHandler::SetArchiveContent(std::string text_content,
+                                            bool is_video) {
+  // Construct a "content archive" implementation of AssociatedContentDelegate
+  // with a duplicate of the article text.
+  auto archive_content = std::make_unique<AssociatedArchiveContent>(
+      metadata_->associated_content->url.value_or(GURL()),
+      std::move(text_content),
+      base::UTF8ToUTF16(metadata_->associated_content->title.value_or("")),
+      is_video);
+  associated_content_delegate_ = archive_content->GetWeakPtr();
+  archive_content_ = std::move(archive_content);
+  should_send_page_contents_ = true;
```

Comment:
I left it up to the caller since it's called from the constructor once and the other call location calls `OnAssociatedContentInfoChanged` afterwards but only calls `SetArchiveContent` conditionally. I suppose this can be re-worked to read nicer, especially because calling `OnAssociatedContentInfoChanged` from the constructor won't do anything anyway (no clients yet).